probably/pgcl/cfg.py

```diff
@@ -284,7 +284,8 @@ def to_networkx(self) -> nx.DiGraph:
             g.add_node(block.ident)
         for block in self:
             for succ_id in block.terminator.successors:
-                g.add_edge(block.ident, succ_id)
+                if(succ_id != None):
```

Comment:
pylint sees two problems here: There are superfluous parentheses and you should use `is not None` instead of `!= None`.

Further, could you explain why you can make this change? If I recall correctly, `None` is actually used as a node name and actually rendered as the `end` node:
https://github.com/Philipp15b/probably/blob/1d5af7af97905d404e535d7315a7b97f1abb2133/probably/pgcl/cfg.py#L333
Doesn't your change simply discard all edges to the `end` node?

Reply:
If we don't do this, the `add_edge` function throws an error when we try to add an edge to `None`, meaning the program crashes (this is in newer versions of the networkx package).

According to the documentation of the function, also in older versions of the package, it was never allowed to add edges to `None`. I don't know how older versions handled this, but like I said, newer ones throw an error. So we either need to specify that probably uses an older version of networkx, if that allows edges to `None` even though the documentations states otherwise, or somehow forbid `None` from being used here.

Reply:
I think we should convert `None` nodes to some other value like `end` for networkx. Maybe even clean up the `cfg` module so that it doesn't internally use `None` but some other value.

Why was networkx upgraded? For the moment, we could stick to an older version and fix the problem later.

Reply:
> Why was networkx upgraded?
This is a result of b2ede13b8b02c619a3c2a2adbae355779accc914.

Reply:
This is in the changelog, so I'll resolve it here